util/template/expression_template.go

```diff
@@ -61,6 +61,9 @@ func expressionReplace(w io.Writer, expression string, env map[string]interface{
 	if (err != nil || result == nil) && allowUnresolved {
 		//  <nil> result is also un-resolved, and any error can be unresolved
 		log.WithError(err).Debug("Result and error are unresolved")
+		if err != nil && strings.Contains(err.Error(), "interface conversion") {
+			return 0, fmt.Errorf("failed to evaluate expression: %w", err)
+		}
```

Comment:
@agilgur5 not sure if I am a fan of this kind of string based error matching, this could easily break if expr lang got updated. 

Do you have opinions?

I think I'd rather this be refactored so that it's more explicit that it belongs to expr-lang.

Probably util/expr/argoexpr/error.go ?